src/neptune/progress_bar.py

```diff
@@ -0,0 +1,195 @@
+#
+# Copyright (c) 2023, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+__all__ = [
+    "ProgressBarCallback",
+    "TqdmNotebookProgressBar",
+    "TqdmProgressBar",
+    "ClickProgressBar",
+    "IPythonProgressBar",
+    "NullProgressBar",
+]
+
+import abc
+import contextlib
+from types import TracebackType
+from typing import (
+    Any,
+    Callable,
+    Optional,
+    Type,
+)
+
+GENERIC_FUNC_TYPE = Callable[[Any], Any]
+
+
+def _handle_import_error(dependency: str) -> Callable[[GENERIC_FUNC_TYPE], GENERIC_FUNC_TYPE]:
+    def deco(func: GENERIC_FUNC_TYPE) -> GENERIC_FUNC_TYPE:
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            try:
+                return func(*args, **kwargs)
+            except ImportError as e:
+                raise ModuleNotFoundError(
+                    f"Required dependency for progress bar not found. Run 'pip install {dependency}'."
+                ) from e
+
+        return wrapper
+
+    return deco
+
+
+class ProgressBarCallback(contextlib.AbstractContextManager):
```

Comment:
Let's move it more internally and expose it in `neptune.__init__` or `neptune.typing`.